### PR TITLE
Implement branches filter for workflows

### PR DESCRIPTION
**Pull request by danidoni**
See the [developer documentation](https://github.com/openSUSE/open-build-service/wiki/Better-SCM-CI-Integration#branches-filter) for branches filter.

### CODE REVIEW

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

**Comment by danidoni** [comment]
[Implement branches filter](https://trello.com/c/BLeVSxhC/1751-implement-branches-filter)

File: src/api/app/services/trigger_controller_service/scm_extractor.rb [hunk 1]
```diff
@@ -21,18 +21,29 @@ def call
...
+        payload.merge!({ commit_sha: @payload.dig(:pull_request, :head, :sha),
+                         pr_number: @payload[:number],
+                         source_branch: @payload.dig(:pull_request, :head, :ref),
+                         target_branch: @payload.dig(:pull_request, :base, :ref),
+                         action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+                         source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+                         target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name) })
+      when 'push'
+        payload.merge!({ commit_sha: @payload[:after],
+                         branch: @payload[:ref].split('/').last,
```
Comment (hennevogel) [inline src/api/app/services/trigger_controller_service/scm_extractor.rb:40]: branch names can contain slashes too.

Comment (danidoni) [reply src/api/app/services/trigger_controller_service/scm_extractor.rb]: Fixed

Comment (danidoni) [reply src/api/app/services/trigger_controller_service/scm_extractor.rb]: In the end, those changes are landing [here](https://github.com/openSUSE/open-build-service/pull/11641/files) so I've removed them from this PR.

**Review by rubhanazeem — APPROVED** [review]

**Pull request merged**